core/core-impl/src/main/java/ai/timefold/solver/core/impl/domain/variable/listener/support/VariableListenerSupport.java

```diff
@@ -244,7 +271,7 @@ private void simulateGenuineVariableChange(Object entity) {
             if (variableDescriptor.isGenuineListVariable()) {
                 var listVariableDescriptor = (ListVariableDescriptor<Solution_>) variableDescriptor;
                 int size = listVariableDescriptor.getListVariable(entity).size();
-                beforeListVariableChanged(listVariableDescriptor, entity, 0, size);
+                beforeListVariableChangedUnchecked(listVariableDescriptor, entity, 0, size);
```

Comment:
This is a call to artifically trigger all listeners. The check above must be skipped, otherwise this would never pass. It is harmless, as it only exists to update all possible shadow vars.